src/pybind/mgr/prometheus/module.py

```diff
@@ -165,9 +167,25 @@ class HealthHistory:
     def __init__(self, mgr: MgrModule):
         self.mgr = mgr
         self.lock = threading.Lock()
+        self.max_entries = cast(int, self.mgr.get_localized_module_option('healthcheck_history_max_entries', 1000))
+        self.stale_ttl = cast(int, self.mgr.get_localized_module_option('healthcheck_history_stale_ttl', 3600))
         self.healthcheck: Dict[str, HealthCheckEvent] = {}
         self._load()
 
+    def _prune(self, now: float) -> None:
+        """Drop stale/inactive entries and enforce max_entries limit."""
+        stale_keys = [k for k, v in self.healthcheck.items()
+                    if not v.active and (now - v.last_seen) > self.stale_ttl]
+        for k in stale_keys:
+            self.healthcheck.pop(k, None)
+
+        # Enforce max_entries limit
+        if len(self.healthcheck) > self.max_entries:
+            # Sort by last_seen and remove oldest entries
+            sorted_checks = sorted(self.healthcheck.values(), key=lambda x: x.last_seen)
+            for check in sorted_checks[:len(self.healthcheck) - self.max_entries]:
+                self.healthcheck.pop(check.name, None)
```

Comment:
Why not using [`functools` LRU cache](https://docs.python.org/3/library/functools.html#functools.lru_cache) (with `weakref`s)? With that we don't need TTL (weakrefs will be voided on the mgr C++ side), and that cache explicitly tackles the size count.

Reply:
Yeah, I was suggesting that we simplified that struct. What about a `collections.deque`? It's basically a fixed-size FIFO queue. I assume that the events are inserted in order (maybe I'm wrong here), so if that's true, then we don't need an expiration time, just the regular FIFO eviction.

Reply:
Got it!